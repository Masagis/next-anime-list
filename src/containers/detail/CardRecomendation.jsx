import React from 'react'

function CardRecomendation(props) {
  const { data, handleDetail } = props

  return (
    <div
      data-testid="card-recomendation-anime"
      aria-hidden
      className="col-md-2"
      style={{ cursor: 'pointer' }}
      onClick={() => handleDetail()}
    >
      <div className="card shadow mb-3">
        <img
          data-testid="image-recomendation-anime"
          src={data?.entry?.images?.jpg?.image_url}
          alt="banner-anime"
          className="mb-2 img-fluid rounded"
        />
        <p className="text-center" data-testid="title-recomendation-anime">
          {data?.entry?.title}
        </p>
      </div>
    </div>
  )
}

export default CardRecomendation
