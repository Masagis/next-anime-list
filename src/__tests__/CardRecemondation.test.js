/* eslint-disable no-undef */
import { fireEvent, render, screen } from '@testing-library/react'
import CardRecomendation from '../containers/detail/CardRecomendation'

describe('Component CardRecomendation', () => {
  test('should be render the component card recomendation anime by id', () => {
    const data = {
      entry: {
        mal_id: 11061,
        url: 'https://myanimelist.net/anime/11061/Hunter_x_Hunter_2011',
        images: {
          jpg: {
            image_url:
              'https://cdn.myanimelist.net/images/anime/1337/99013.jpg?s=1838e905a0aa3542a009fbcf78000701',
            small_image_url:
              'https://cdn.myanimelist.net/images/anime/1337/99013t.jpg?s=1838e905a0aa3542a009fbcf78000701',
            large_image_url:
              'https://cdn.myanimelist.net/images/anime/1337/99013l.jpg?s=1838e905a0aa3542a009fbcf78000701',
          },
          webp: {
            image_url:
              'https://cdn.myanimelist.net/images/anime/1337/99013.webp?s=1838e905a0aa3542a009fbcf78000701',
            small_image_url:
              'https://cdn.myanimelist.net/images/anime/1337/99013t.webp?s=1838e905a0aa3542a009fbcf78000701',
            large_image_url:
              'https://cdn.myanimelist.net/images/anime/1337/99013l.webp?s=1838e905a0aa3542a009fbcf78000701',
          },
        },
        title: 'Hunter x Hunter (2011)',
      },
      url: 'https://myanimelist.net/recommendations/anime/21-11061',
      votes: 65,
    }

    const handleClickDetail = jest.fn()

    render(<CardRecomendation data={data} handleDetail={handleClickDetail} />)
    expect(screen.getByTestId('card-recomendation-anime')).toBeInTheDocument()
    expect(screen.getByTestId('image-recomendation-anime')).toHaveAttribute(
      'src',
      'https://cdn.myanimelist.net/images/anime/1337/99013.jpg?s=1838e905a0aa3542a009fbcf78000701',
    )
    expect(screen.getByTestId('title-recomendation-anime')).toHaveTextContent(
      'Hunter x Hunter (2011)',
    )

    fireEvent.click(screen.getByTestId('card-recomendation-anime'))
    expect(handleClickDetail).toBeCalledTimes(1)
  })

  test('snapshoot', () => {
    const data = {
      entry: {
        mal_id: 11061,
        url: 'https://myanimelist.net/anime/11061/Hunter_x_Hunter_2011',
        images: {
          jpg: {
            image_url:
              'https://cdn.myanimelist.net/images/anime/1337/99013.jpg?s=1838e905a0aa3542a009fbcf78000701',
            small_image_url:
              'https://cdn.myanimelist.net/images/anime/1337/99013t.jpg?s=1838e905a0aa3542a009fbcf78000701',
            large_image_url:
              'https://cdn.myanimelist.net/images/anime/1337/99013l.jpg?s=1838e905a0aa3542a009fbcf78000701',
          },
          webp: {
            image_url:
              'https://cdn.myanimelist.net/images/anime/1337/99013.webp?s=1838e905a0aa3542a009fbcf78000701',
            small_image_url:
              'https://cdn.myanimelist.net/images/anime/1337/99013t.webp?s=1838e905a0aa3542a009fbcf78000701',
            large_image_url:
              'https://cdn.myanimelist.net/images/anime/1337/99013l.webp?s=1838e905a0aa3542a009fbcf78000701',
          },
        },
        title: 'Hunter x Hunter (2011)',
      },
      url: 'https://myanimelist.net/recommendations/anime/21-11061',
      votes: 65,
    }

    const renderComponent = render(<CardRecomendation data={data} />)
    expect(renderComponent).toMatchSnapshot()
  })
})
