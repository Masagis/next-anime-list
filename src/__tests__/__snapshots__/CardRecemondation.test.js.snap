// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CardRecomendation snapshoot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-hidden="true"
        class="col-md-2"
        data-testid="card-recomendation-anime"
        style="cursor: pointer;"
      >
        <div
          class="card shadow mb-3"
        >
          <img
            alt="banner-anime"
            class="mb-2 img-fluid rounded"
            data-testid="image-recomendation-anime"
            src="https://cdn.myanimelist.net/images/anime/1337/99013.jpg?s=1838e905a0aa3542a009fbcf78000701"
          />
          <p
            class="text-center"
            data-testid="title-recomendation-anime"
          >
            Hunter x Hunter (2011)
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-hidden="true"
      class="col-md-2"
      data-testid="card-recomendation-anime"
      style="cursor: pointer;"
    >
      <div
        class="card shadow mb-3"
      >
        <img
          alt="banner-anime"
          class="mb-2 img-fluid rounded"
          data-testid="image-recomendation-anime"
          src="https://cdn.myanimelist.net/images/anime/1337/99013.jpg?s=1838e905a0aa3542a009fbcf78000701"
        />
        <p
          class="text-center"
          data-testid="title-recomendation-anime"
        >
          Hunter x Hunter (2011)
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
